src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -0,0 +1,696 @@
+            foreach (Http3RequestStream stream in streamsToGoAway)
+            {
+                stream.GoAway();
+            }
+        }
+
+        public void RemoveStream(long streamId)
+        {
+            lock (SyncObj)
+            {
+                bool removed = _activeRequests.Remove(streamId);
+                Debug.Assert(removed == true);
+
+                if (ShuttingDown)
+                {
+                    CheckForShutdown();
+                }
+            }
+        }
+
+        public override void Trace(string message, [CallerMemberName] string memberName = null) =>
+            Trace(0, message, memberName);
+
+        internal void Trace(long streamId, string message, [CallerMemberName] string memberName = null) =>
+            NetEventSource.Log.HandlerMessage(
+                _pool?.GetHashCode() ?? 0,    // pool ID
+                GetHashCode(),                // connection ID
+                (int)streamId,                // stream ID
+                memberName,                   // method name
+                message);                     // message
+
+        private async ValueTask SendSettingsAsync()
+        {
+            try
+            {
+                _clientControl = _connection.OpenUnidirectionalStream();
+
+                var buffer = new byte[12];
+
+                int payloadLength = 1 + VariableLengthIntegerHelper.WriteInteger(buffer.AsSpan(3), _pool.Settings._maxResponseHeadersLength * 1024);
+                buffer[0] = 0x00; // Control stream type identifier.
+                buffer[1] = (byte)Http3FrameType.Settings;
+                buffer[2] = (byte)payloadLength;
+                buffer[3] = (byte)Http3SettingType.MaxHeaderListSize;
+
+                await _clientControl.WriteAsync(buffer.AsMemory(4 + payloadLength), CancellationToken.None).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Abort(ex);
+            }
+        }
+
+        /// <summary>
+        /// Accepts unidirectional streams (control, QPack, ...) from the server.
+        /// </summary>
+        private async Task AcceptStreamsAsync()
+        {
+            try
+            {
+                while (true)
+                {
+                    ValueTask<QuicStream> streamTask;
+
+                    lock (SyncObj)
+                    {
+                        if (_connection == null || ShuttingDown)
+                        {
+                            return;
+                        }
+
+                        // No cancellation token is needed here; we expect the operation to cancel itself when _connection is disposed.
+                        streamTask = _connection.AcceptStreamAsync(CancellationToken.None);
+                    }
+
+                    QuicStream stream = await streamTask.ConfigureAwait(false);
+                    _ = ProcessServerStreamAsync(stream);
```

Comment:
I'd keep this task around and await it in some cleanup code.

Reply:
There's no need here; the processed is cleaned up when the connection is disposed, and errors are only observed through `SendAsync()`. Will add a comment though.